### PR TITLE
Documentation changes

### LINKED ISSUE #133: Create DeltaCode documentation on ReadTheDocs
The primary DeltaCode documentation is currently at the top of the AboutCode website on ReadtheDocs (RTD). It should be moved to its own RTD website. This task will also:
- Include cross-checking it with the Wiki documentation and then removing the Wiki pages
- Versioning the DeltaCode RTD website in sync with the code.

### PR DESCRIPTION
**Pull request by pratik0316**
This  PR is meant for updating the documentation at various locations in the codebase, as well as adding/updating an extensive docs in `ReadTheDocs` owing to the changes due to integration of the `VirtualCodebase`

Signed-off-by: Pratik Dey <pratikrocks.dey11@gmail.com>

### CODE REVIEW

**Review by steven-esser — APPROVED** [review]

**Review by steven-esser — CHANGES_REQUESTED** [review]
I think this just needs to be rebased with main; but this can be merged right after.

**Comment by pratik0316** [comment]
Yes we need to rebase it , and some modifications are required to in this PR

File: docs/source/deltacode/comprehensive_installation.rst [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+- The installation is complete.
+
+Un-installation
+---------------
+
+- Delete the directory in which you extracted DeltaCode.
+- Delete any temporary files created in your system temp directory under a deltacode directory.
+
+Using the docker image for testing Deltacode
```
Comment (pratik0316) [inline docs/source/deltacode/comprehensive_installation.rst:124]: This is the required instructions to run the docker image

**Comment by AyanSinhaMahapatra** [comment]
@Pratikrocks does this replace the absolete docs at https://github.com/nexB/aboutcode/tree/master/docs/source/deltacode?

**Comment by pratik0316** [comment]
Yes it replaces the old one

**Comment by pratik0316** [comment]
Though the GSOC work report(.rst) is not present here

**Comment by AyanSinhaMahapatra** [comment]
Right, then you could open a PR deleting them at https://github.com/nexB/aboutcode/tree/master/docs/source/deltacode and [here](https://github.com/nexB/aboutcode/blob/master/docs/source/index.rst#aboutcode-project-docs) link to the new docs.

The GSoC report can be added here in this PR (in deltacode RTD) also.

**Comment by pratik0316** [comment]
Alright I am doing it

File: docs/source/deltacode/GSOC_2021_report.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
+Google Summer of Code 2021 - Final report
```
Comment (pratik0316) [inline docs/source/deltacode/GSOC_2021_report.rst:1]: added the .rst here @AyanSinhaMahapatra

Comment (AyanSinhaMahapatra) [reply docs/source/deltacode/GSOC_2021_report.rst:1]: LGTM

**Review by pombredanne — APPROVED** [review]

**Pull request merged**